framework/src/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -946,15 +947,17 @@ public Void scan(AnnotatedTypeMirror t, AnnotationMirror qual) {
                     case IMPLICIT_UPPER_BOUND:
```

Comment:
This isn't your fault, but I found the lack of pre-existing documentation in the class made this code review difficult to perform.  For example, the location field that is the expression for this case statement isn't documented.  What is the relationship between apply and doApply?  (Neither has documentation.)  Could you improve the documentation, either in this pull request or in another one that we can apply before this one?